router/broker.go

```diff
@@ -147,13 +154,14 @@ func (b *Broker) Publish(pub *session, msg *wamp.Publish) {
 				Details: wamp.Dict{},
 				Error:   wamp.ErrOptionDisallowedDiscloseMe,
 			})
+			return
 		}
 		disclose = true
 	}
 	pubID := wamp.GlobalID()
 
 	// Get blacklists and whitelists, if any, from publish message.
-	filter := newPublishFilter(msg)
+	filter := b.filterFactory(msg)
```

Comment:
This step was done for the particular filter implementation I was using, to "pre-compile" the block-whitelists into maps.  I am not sure this is the right approach for letting users provide their own filters.   I need to think about this.  

If this is the right approach then need to document/encourage implementer to do as much much work as possible in the FilterFactory function, and not in the filter itself (since the filter is executed in the broker's critical section).

In general, I want to avoid making it possible for the user to define something that is executed in a broker or dealer critical section (single core goroutine that accesses mutable state).  If the user needs to supply some functionality, it should be runable in the session's message handling goroutine, not in core of broker or dealer.  Since matching against subscriber must be done in core, every effort must be taken to minimize what is done there.  I think many users trying to use the feature will end up slowing down their router and not know why.

**An Alternative Approach**
An alternative approach is to create a client-side filter tool.  This uses the session meta API to get information about all sessions, and to keep that info updated.  When the client wants to publish a message, it runs its client filter and selects a list of session IDs it wants to publish to, it then puts these IDs into a simple whitelist and publishes the message.

Consider the performance advantage of the above.  Suppose there are 500 clients trying to publish messages to a filtered set of subscribers .  The client-side tool would distribute most of the filter processing over the clients.  Also, the filtering could be as complex and slow as needed -- consider looking up security policy and ACLs in a database to decide who to publish to.